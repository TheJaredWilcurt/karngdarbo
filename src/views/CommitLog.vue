<template>
  <div class="commitlog">
    Commit Log.
    <button @click="openDevTools">
      Open Dev tools
    </button>
    <button @click="getBranches">
      Get Branches
    </button>
    <input v-model="newRepoPath" type="text" @keyup.enter="setNewRepoPath" />
    <div>
      <strong>App Loading</strong>: <pre>{{ appLoading }}</pre>
    </div>
    <div>
      <strong>App Error</strong>: <pre>{{ appError }}</pre>
    </div>
    <div>
      <strong>Repo Path</strong>: <pre>{{ repoPath }}</pre>
    </div>
    <div>
      <strong>Branches</strong>: <pre>{{ branches }}</pre>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';
export default {
  name: 'CommitLog',
  data: function () {
    return {
      newRepoPath: ''
    };
  },
  methods: {
    openDevTools: function () {
      window.nw.Window.get().showDevTools();
    },
    getBranches: function () {
      this.$store.dispatch('getBranchList');
    },
    setNewRepoPath: function () {
      this.$store.commit('setRepoPath', this.newRepoPath);
      this.getBranches();
    }
  },
  computed: {
    ...mapState([
      'appLoading',
      'appError',
      'branches',
      'repoPath'
    ])
  }
};
</script>
