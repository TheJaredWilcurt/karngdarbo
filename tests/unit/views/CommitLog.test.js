import { shallowMount } from '@vue/test-utils';

import CommitLog from '@/views/CommitLog.vue';

describe('CommitLog.vue', () => {
  describe('Created', () => {
    test('Default snapshot', () => {
      const wrapper = shallowMount(CommitLog);

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });
});
