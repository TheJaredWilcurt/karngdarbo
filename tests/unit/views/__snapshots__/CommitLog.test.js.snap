// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitLog.vue Created Default snapshot 1`] = `
<div class="commitlog">
  Commit Log.
</div>
`;
